### ISSUE TITLE
error while "composer update"

### ISSUE THREAD

**Issue by SidneyKim** [composer/composer#6048]
Thanks in advance.

My `composer.json`:

```json
{
  "name": "KETSE/casebox",
  "license": "AGPL",
  "type": "project",
  "config": {
    "preferred-install": "dist",
    "autoloader-suffix": "Casebox",
    "github-protocols": ["https"],
    "process-timeout": 600
  },
  "autoload": {
    "psr-4": {
      "": "src/"
    },
    "classmap": [
      "app/AppKernel.php",
      "app/AppCache.php",
      "app/AppEnv.php"
    ]
  },
  "require": {
    "php": ">=5.5.9",
    "symfony/symfony": "3.0.*",
    "doctrine/orm": "^2.5",
    "doctrine/doctrine-bundle": "^1.6",
    "doctrine/doctrine-cache-bundle": "^1.2",
    "symfony/swiftmailer-bundle": "^2.3",
    "symfony/monolog-bundle": "^2.8",
    "symfony/translation": "*",
    "sensio/distribution-bundle": "^5.0",
    "sensio/generator-bundle": "^3.0",
    "symfony/doctrine-bridge": "^3.0",
    "sensio/framework-extra-bundle": "^3.0.2",
    "kwi/urllinker": "@dev"
  },
  "require-dev": {
    "caseboxdev/core-bundle": "@dev",
    "caseboxdev/rpc-bundle": "@dev",
    "caseboxdev/rest-bundle": "@dev",
    "phpunit/phpunit": "^5.2.12",
    "phpunit/php-code-coverage": "^3.3.0",
    "satooshi/php-coveralls": "dev-master",
    "squizlabs/php_codesniffer": "^2.6",
    "escapestudios/symfony2-coding-standard": "^2.9",
    "phpmd/phpmd": "^2.4"
  },
  "scripts": {
    "post-install-cmd": [
      "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget",
      "Casebox\\CoreBundle\\Composer\\ScriptHandler::buildAssets"
    ],
    "post-update-cmd": [
      "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget",
      "Casebox\\CoreBundle\\Composer\\ScriptHandler::buildAssets",
      "Casebox\\CoreBundle\\Composer\\ScriptHandler::buildTranslations"
    ]
  },
  "extra": {
    "symfony-app-dir": "app",
    "symfony-bin-dir": "bin",
    "symfony-var-dir": "var",
    "symfony-web-dir": "web",
    "symfony-tests-dir": "tests",
    "symfony-assets-install": "relative",
    "incenteev-parameters": {
      "file": "app/config/default/parameters.yml"
    }
  }
}
```

Output of `composer diagnose`:

```
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Checking composer.json: WARNING
License "AGPL" is not a valid SPDX license identifier, see https://spdx.org/licenses/ if you use an open license.
If the software is closed-source, you may use "proprietary" as license.
Defining autoload.psr-4 with an empty namespace prefix is a bad idea for performance
require.symfony/translation : unbound version constraints (*) should be avoided
require.kwi/urllinker : unbound version constraints (@dev) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer update
```

I get the following output:

```
{

  "name": "KETSE/casebox",

  "license": "AGPL",

  "type": "project",

  "config": {

    "preferred-install": "dist",

    "autoloader-suffix": "Casebox",

    "github-protocols": ["https"],

    "process-timeout": 600

  },

  "autoload": {

    "psr-4": {

      "": "src/"

    },

    "classmap": [

      "app/AppKernel.php",

      "app/AppCache.php",

      "app/AppEnv.php"

    ]

  },

  "require": {

    "php": ">=5.5.9",

    "symfony/symfony": "3.0.*",

    "doctrine/orm": "^2.5",

    "doctrine/doctrine-bundle": "^1.6",

    "doctrine/doctrine-cache-bundle": "^1.2",

    "symfony/swiftmailer-bundle": "^2.3",

    "symfony/monolog-bundle": "^2.8",

    "symfony/translation": "*",

    "sensio/distribution-bundle": "^5.0",

    "sensio/generator-bundle": "^3.0",

    "symfony/doctrine-bridge": "^3.0",

    "sensio/framework-extra-bundle": "^3.0.2",

    "kwi/urllinker": "@dev"

  },

  "require-dev": {

    "caseboxdev/core-bundle": "@dev",

    "caseboxdev/rpc-bundle": "@dev",

    "caseboxdev/rest-bundle": "@dev",

    "phpunit/phpunit": "^5.2.12",

    "phpunit/php-code-coverage": "^3.3.0",

    "satooshi/php-coveralls": "dev-master",

    "squizlabs/php_codesniffer": "^2.6",

    "escapestudios/symfony2-coding-standard": "^2.9",

    "phpmd/phpmd": "^2.4"

  },

  "scripts": {

    "post-install-cmd": [

      "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget",

      "Casebox\\CoreBundle\\Composer\\ScriptHandler::buildAssets"

    ],

    "post-update-cmd": [

      "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",

      "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget",

      "Casebox\\CoreBundle\\Composer\\ScriptHandler::buildAssets",

      "Casebox\\CoreBundle\\Composer\\ScriptHandler::buildTranslations"

    ]

  },

  "extra": {

    "symfony-app-dir": "app",

    "symfony-bin-dir": "bin",

    "symfony-var-dir": "var",

    "symfony-web-dir": "web",

    "symfony-tests-dir": "tests",

    "symfony-assets-install": "relative",

    "incenteev-parameters": {

      "file": "app/config/default/parameters.yml"

    }

  }

}
```

And I expected this to happen:
working

**Comment by SidneyKim** [comment]
I solved it by adding --dev at the end of the command.
composer update --dev

![image](https://cloud.githubusercontent.com/assets/6936069/21711416/615fbdba-d432-11e6-9bca-0d1d0953fe0a.png)

**Comment by alcohol** [comment]
Your "output" is just the copy and paste of your composer.json again... seems like you solved it though, whatever "it" was.

**Comment by stof** [comment]
Well, I suspect that the issue is the usage of Symfony 3.0.x together with the newly released Twig 2, as Symfony 3.0.x is not compatible with the final version of Twig 2.x (it reached end of maintenance before the last BC breaks of Twig 2.x).
Btw, a better fix is to use ``"twig/twig": "^1.25",`` rather than forcing exactly the 1.25.0 release, so that you get the next 1.x releases when they are done @SidneyKim
